### PR TITLE
[macOS] Nuke XBMCHelper

### PR DESCRIPTION
**Pull request by enen92**
## Description
XBMCHelper is a little tool that allows Kodi to be controller via old Apple Remotes (those that date back the Siri remote - using the builtin infrared receiver https://en.wikipedia.org/wiki/Apple_Remote). It includes other stuff like a builtin Launch Agent and the companion event server remote.

The last mac models featuring the IR receiver (and the companion apple remote) are the MacBook Pro mid 2012 model and the mac mini mid 2012. The last macOS version allowed to run on such equipment is Mojave (10.14), precisely the minimum version required to run Kodi Omega.
Since it's not a good policy to simply drop stuff that might still be supported and usable (and used?), I propose to drop this once the macOS minimum requirement is bumped - rendering the last devices featuring the IR remote obsolete for Kodi. I honestly doubt we still have any users using such equipment for Kodi but you never know - this might even be easily (or already) supported in peripheral.joystick - I don't have any hardware to confirm.
Recent macs need some external usb IR receivers to still use those remotes (plus the required adaptation software - e.g. flirc) so this doesn't really make sense for recent macs (especially those with apple silicon).

edit: looks like this info is wrong, please read https://github.com/xbmc/xbmc/pull/23580#issuecomment-3703529047 and https://github.com/xbmc/xbmc/pull/23580#issuecomment-3703594784 

## Motivation and context
Cleanup

## What is the effect on users?
Hopefully none, ~Kodi must support macOS > 10.14 when this is merged.~

### CODE REVIEW

**Comment by enen92** [comment]
jenkins build this please

**Comment by enen92** [comment]
Jenkins build this please

**Comment by enen92** [comment]
Looking into my old PRs it seems I was wrong after all...
Underneath Kodi uses https://www.iospirit.com/labs/hidremote/ which states the following:

_"The HIDRemote Objective-C class provides your application with access to the Apple Remote IR Receiver under **OS X 10.4 (Tiger) up to OS X 10.13 (High Sierra).** It is available under a BSD-style license."_

This also seems to be correct and not just outdated documentation:
What changed after High Sierra (10.14+)

Starting with macOS 10.14 Mojave, Apple:

**1. Removed IR hardware support**

New Macs no longer include IR receivers.

**2. Removed or disabled AppleIRController**

The kernel extension either:

No longer exists, or

Is blocked by System Integrity Protection (SIP), or

No longer publishes HID events

**3. Hardened IOKit + HID access**

Apple tightened access to:

Private IOKit classes

Event taps

HID injection paths

So even if you:

- Compile HIDRemote successfully
- Run it on modern macOS

It will not receive any IR events, because the OS never generates them anymore.

**tldr:** this is broken, doesn't work since high sierra and can be safely removed from Kodi today

**Comment by enen92** [comment]
@claude can you confirm my last comment is correct and that xbmchelper is safe to drop considering our last supported macOS version is Mojave (10.14)?

**Comment by enen92** [comment]
@claude can you check the last revision please?
Did I forget some place on the cleanup or left some unused include?

**Comment by enen92** [comment]
Jenkins build this please

**Review by kambala-decapitator — APPROVED** [review]
the change itself is fine

> The last macOS version allowed to run on such equipment is Mojave (10.14)

it's not entirely true: one can unofficially run all the later OS versions via OCLP (only don't know about 26)

**Comment by fuzzard** [comment]
Also from memory, didn't the bump to cpython 3.14 also force our minimum version to 10.15?

**Comment by enen92** [comment]
> the change itself is fine
> 
> > The last macOS version allowed to run on such equipment is Mojave (10.14)
> 
> it's not entirely true: one can unofficially run all the later OS versions via OCLP (only don't know about 26)

You can but the OS won't expose the events anyway, so this is dead code

**Comment by enen92** [comment]
> Also from memory, didn't the bump to cpython 3.14 also force our minimum version to 10.15?

From what I could search that only affects python officially distributed packages, don't really put any hard requirement for self compilation. But I am not sure, I don't see anything on that PR that forces 10.15. if that's true our docs are outdated (also we have a few TODOs to drop stuff when we get to 10.15)

**Comment by fuzzard** [comment]
Id say it's more our docs are outdated, but someone would need to trawl through the cpyrhon stuff to work out if it's actually a hard requirement, so I'm happy to take the comment mentioning it at face value
